code/game/gamemodes/nuclear/pinpointer.dm

```diff
@@ -102,17 +135,29 @@
 		if(mode == 2)
 			point_at(target)
 		to_chat(usr, "<span class='notice'>You activate the pinpointer.</span>")
+		START_PROCESSING(SSfastprocess, src)
 	else
 		active = 0
 		icon_state = icon_off
 		to_chat(usr, "<span class='notice'>You deactivate the pinpointer.</span>")
+		STOP_PROCESSING(SSfastprocess, src)
+
+/obj/item/pinpointer/advpinpointer/process()
+	if(active == 1)
+		if(mode == 0)
+			workdisk()
+		if(mode == 1)
+			point_at(location)
+		if(mode == 2)
+			point_at(target)
+	else
+		icon_state = icon_off
```

Comment:
I don't think you have to update the icon every tick. Just when you turn it off it should be updated

Reply:
removed